src/components/chart-preview/ChartPreview.tsx

```diff
@@ -52,6 +55,42 @@ export const ActualChart = ({
     useSaveCsvData(csv, "chart-data");
   };
 
+  const onFigureDownloadClick = (e: any) => {
+    //TODO rework getting parent node
+    const fullFigureNode =
+      e.target.parentNode.parentNode.parentNode.parentNode.parentNode;
```

Comment:
Discussed: try getting to the highest chartbuilder block and do one parentNode for the figure

Reply:
added a function to get the top level node of chartBuilder then get the parent parentNode, which will be the full figure.